pkg/apiserver/domain/service/project.go

```diff
@@ -495,7 +495,11 @@ func (p *projectServiceImpl) GetConfigs(ctx context.Context, projectName, config
 		// legacy providers
 		var providers = &terraformapi.ProviderList{}
 		if err := p.K8sClient.List(ctx, providers, client.InNamespace(types.DefaultAppNamespace)); err != nil {
-			return nil, err
+			// this logic depends on the terraform addon, ignore the no matches kind error before the terraform addon is installed.
+			if !strings.Contains(err.Error(), "no matches for kind \"Provider\" in version \"terraform.core.oam.dev") {
```

Comment:
`meta.IsNoMatchError(err)` Replace with this code is more clear?

Reply:
> `meta.IsNoMatchError(err)` Replace with this code is more clear?

Thanks. Updated.